support-lib/java/com/snapchat/djinni/Future.java

```diff
@@ -0,0 +1,107 @@
+/**
+  * Copyright 2021 Snap, Inc.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package com.snapchat.djinni;
+
+public class Future<T> {
+
+    public interface FutureHandler<U> {
+        public void handleResult(Future<U> res) throws Throwable;
+    }
+
+    public interface FutureHandlerWithReturn<U, R> {
+        public R handleResult(Future<U> res) throws Throwable;
+    }
+    
+    private SharedState<T> sharedState;
+
+    Future(SharedState<T> state) {
+        sharedState = state;
+    }
+
+    public boolean isReady() {
+        synchronized(sharedState) {
+            return sharedState.isReady();
+        }
+    }
+
+    public T get() throws Throwable {
```

Comment:
does need to be declared as throwing?

Reply:
calling `get()` will throw the stored exception.